src/js/actions/export.js

```diff
@@ -461,16 +470,30 @@ define(function (require, exports) {
             return Promise.resolve("no assets to export");
         }
 
-        // Iterate over the exports map, find the associated layer, test of "exportEnabled"
-        var exportArray = documentExports.getLayersWithExports(document, undefined, true)
-            .flatMap(function (layer) {
-                return documentExports.getLayerExports(layer.id)
-                    .map(function (asset, index) {
-                        return _exportAsset.call(this, document, layer, index, asset);
-                    }, this);
-            }, this);
-
-        return Promise.all(exportArray.toArray());
+        return _exportService.promptForFolder(_lastFolderPath || "~")
+            .bind(this)
+            .then(
+                function (baseDir) {
+                    _lastFolderPath = baseDir;
+
+                    // Iterate over the exports map, find the associated layer, test of "exportEnabled"
+                    var exportArray = documentExports.getLayersWithExports(document, undefined, true)
+                        .flatMap(function (layer) {
+                            return documentExports.getLayerExports(layer.id)
+                                .map(function (asset, index) {
+                                    return _exportAsset.call(this, document, layer, index, asset, baseDir);
+                                }, this);
+                        }, this);
+
+                    return Promise.all(exportArray.toArray());
+                },
+                function (err) {
+                    // promptForFolder rejected, probably just user-canceled
+                    if (err.message && err.message.startsWith("cancelError: cancel")) {
+                        log.warn("Prompt for folder failed, and it wasn't a simple 'cancel'");
+                    }
+                    return Promise.resolve();
```

Comment:
I feel that since the JSDoc makes no mention of what is being promised, it is OK as is.  I updated the JSDoc to be slightly more clear.  This promise gets dumped on the floor anyway.